modules/standard/IO.chpl

```diff
@@ -5228,6 +5228,69 @@ proc _channel.writeBinary(arg:numeric, endian:ioendian) throws {
   }
 }
 
+/*
+   Read up to ``maxBytes`` bytes from a fileReader into a :class:`~CTypes.c_ptr`
+
+   Note that native endianess is always used.
+
+   If ``maxBytes`` is not evenly divisible by the size of ``t``, then the
+   remaining bytes are ignored.
+
+   :arg ptr: a :class:`~CTypes.c_ptr` to some memory — existing values will be
+              overwritten
+   :arg maxBytes: the maximum number of bytes to read from the fileReader
+   :returns: the number of bytes that were read. can be less than ``maxBytes``
+              if EOF was reached before reading the specified number of bytes,
+              or if ``maxBytes`` is not evenly divisible by the size of ``t``
+
+   :throws SystemError: Thrown if an error occurred while reading from the fileReader
+*/
+proc fileReader.readBinary(ptr: c_ptr(?t), maxBytes: int): int throws {
+  var e: errorCode = 0;
+  const tSize = c_sizeof(t),
+        numToRead = maxBytes / tSize;
+
+  for n in 0..<numToRead {
+    e = try _read_binary_internal(this._channel_internal, iokind.native, ptr[n]);
```

Comment:
This is OK but IMO it should just call `qio_channel_read`. It should improve performance to do so. (But you can check). You could change it now or leave a comment indicating that should happen. If you change it, you might want to compute `len` as `numToRead * tSize` (in other words, maxBytes but leaving out any partial element).